### ISSUE TITLE
Doesn't work nice with sortables / ng-repeat.

### ISSUE THREAD

**Issue by kilianc** [angular-ui/ui-slider#41]
`Uncaught Error: cannot call methods on slider prior to initialization; attempted to call method 'destroy'`

The problem seems to be that after the first time it doesn't init the slider anymore

**Comment by meredithmmyers** [comment]
+1 for a fix for this, I'm running into the same issue with Angular v1.3.5

**Comment by pglotov** [comment]
It may not be a problem with ui-slider, e.g. ui-modal destroys DOM before $destroy'ing scope:
https://github.com/angular-ui/bootstrap/issues/3231

**Comment by fanzeyi** [comment]
An interesting discover:

**Tl;DR: This problem is relevant to the order of executing the dependency libraries.**

Here is my experiment:

I wrote a simple version of this problem on Plunker. [The working version](http://plnkr.co/edit/S87TBUWFsfS0HEdLp39H?p=preview). 

It works fine. But this is actually what I'm using in my project.

I start cut the code in my project that has the problem. I removed everything except the code relevant to the ui-slider section. Surprisingly, the error is still there.

At first, I think it's the version difference caused the problem. I start to change the dependencies version to the plunker gave to me. But the problem is still there.

After I changed the ordering of link libraries. I made the problem occurred. [The broken version](http://plnkr.co/edit/U8pfwQBUzhsSN76ba5qre?p=preview)

I guess it's something related to the ordering of firing the destroy event. 

As a result, just adjust the order of your code.

![screenshot 2015-03-07 15 56 42](https://cloud.githubusercontent.com/assets/409951/6543820/8d4dac86-c4e2-11e4-944f-1f33ae31740f.png)

**Comment by fanzeyi** [comment]
A note for wiredep user:

You can add jquery-ui as a dependency of angularjs to force wiredep insert jquery-ui before angularjs in order to solve this issue.

Gruntfile.js:

```javascript
        overrides: {
            'angular': {
                "dependencies": {
                    "jquery-ui": "~1.11.2"
                }
            }
        }
```

**Comment by pglotov** [comment]
@fanzeyi You 'broken version' shows a 'hello plunker' message in index.html, has empty script.js and has no sliders in it. What is is supposed to prove?

**Comment by fanzeyi** [comment]
@pglotov Sorry it's may because I forget to save the plunker. 

Use this: http://plnkr.co/edit/JTfXu3pHSbQFJ1iLpMQJ?p=preview

and click Second.

**Comment by fanzeyi** [comment]
And in order to prove the connection between the order and ng-repeat which this issue addressed.

http://plnkr.co/edit/Fj03pltXDHi0wCF7jGFs?p=preview

This plunker works fine with ng-repeat with the "right" order.

**Comment by fanzeyi** [comment]
To be clear: I just give a dirty solution and show a potential reason for this error. This doesn't mean I solved this issue.

**Comment by PowerKiKi** [comment]
According to some of you #55 should fix this issue. So closing this, but feel free to re-open if something still mis-behave.

**Comment by samy100** [comment]
@PowerKiKi This still happens to me when using slider inside a ui-modal, any solutions? Seems that the dom is fully destroyed before the scope on modals. The proper way on ui-slider might be to try catch the slider destroy on both $element.$on("$destroy") event and on $scope.$on("$destroy") event to achieve full compatibility. Right now you're destroying the slider on $scope's event and that doesn't work with ui-modal. See https://github.com/angular-ui/bootstrap/issues/3231
Would love to see a fix.

**Comment by PowerKiKi** [comment]
@samy100, a new fix landed in #78
